File: two-pointers/125-valid-palindrome/solution.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        cleaned =  ''

        for x in s:
            if (x >= 'a' and x <= 'z') or (x >='A' and x <='Z') or (x >='0' and x <='9'):
                cleaned = cleaned +x.lower()
        return cleaned == cleaned[::-1]

        # Add this block to run the code locally in VS Code
if __name__ == "__main__":
    solver = Solution()
    # Replace with the test cases from the LeetCode problem
    result = solver.isPalindrome('A man, a plan, a canal: Panama')
    print(f"Output: {result}")

    result = solver.isPalindrome( " " )
    print(f"Output: {result}")

    result = solver.isPalindrome('race a car')
    print(f"Output: {result}")

File: two-pointers/125-valid-palindrome/solution.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isPalindrome = function (s) {

    if (!s || s.length < 0) {
        return false;
    }

    // const cleaned = s.split('').filter((char) => {
    //     const code = char.charCodeAt(0);
    //     return (
    //         (code > 47 && code < 58) || // numeric (0-9)
    //         (code > 64 && code < 91) || // upper alpha (A-Z)
    //         (code > 96 && code < 123)   // lower alpha (a-z)
    //     );
    // }).join('').toLowerCase();

    // if(cleaned === [...cleaned].reverse().join('')) {
    //     return true;
    // }

    let cleaned = '';

    for (let x = 0; x < s.length; x++) {

        if (
            (s[x] >= 'a' && s[x] <= 'z') || // Lowercase
            (s[x] >= 'A' && s[x] <= 'Z') || // Uppercase
            (s[x] >= '0' && s[x] <= '9')    // Numbers
        ) {
            cleaned += s[x].toLowerCase();
        }
    }

    if (cleaned === [...cleaned].reverse().join('')) {
        return true;
    }
    return false;
};

console.log(isPalindrome('A man, a plan, a canal: Panama'));
console.log(isPalindrome('race a car'));
console.log(isPalindrome( " "));